src/platform/backends/qemu/qemu_virtual_machine.cpp

```diff
@@ -380,7 +380,7 @@ void mp::QemuVirtualMachine::shutdown(bool force)
             mpl::log(mpl::Level::debug, vm_name, "No process to kill");
         }
 
-        if (state == State::suspended || mp::backend::instance_image_has_snapshot(desc.image.image_path, suspend_tag))
+        if (state == State::suspended && mp::backend::instance_image_has_snapshot(desc.image.image_path, suspend_tag))
```

Comment:
Correct me if I'm wrong, but wouldn't always deleting the suspend snapshot be the correct behaviour for force stop? Force stop is supposed to "wipe the slate clean", so getting rid of the snapshot seems like the expected result.

```suggestion
        if (mp::backend::instance_image_has_snapshot(desc.image.image_path, suspend_tag))
```

Reply:
That is a very good question actually. I think the state transition of VM guarantees that the image should only have the suspend_tag when the VM is in suspended state. To be more specific, the suspend operation from a running vm creates the suspend_tag, both start and force-stop consume the suspend_tag. 

However, this change might not make a difference in practice, but definitely make things more robust in case the user gets the VM into some unexpected situation.